python/ray/tests/test_gpu_objects_gloo.py

```diff
@@ -24,10 +25,60 @@ def get_gpu_object(self, obj_id: str):
         )
         if gpu_object_store.has_gpu_object(obj_id):
             gpu_object = gpu_object_store.get_gpu_object(obj_id)
-            print(f"gpu_object: {gpu_object}")
             return gpu_object
         return None
 
+    def get_num_gpu_objects(self):
+        gpu_object_manager = ray._private.worker.global_worker.gpu_object_manager
+        return len(gpu_object_manager.gpu_object_store.gpu_object_store)
+
+
+@pytest.mark.parametrize("data_size_bytes", [100])
+def test_gc_gpu_object(ray_start_regular, data_size_bytes):
```

Comment:
Can you also add a test that deletes ref1 before the receiver task has finished?

Reply:
update https://github.com/ray-project/ray/pull/53911/commits/48d4eed990e72079111e66804ab81f8ac717c0af